File: loquacious/src/main/java/com/saantiaguilera/loquacious/util/Mangler.java
package com.saantiaguilera.loquacious.util;

/**
 * Created by saguilera on 11/18/17.
 */
public final class Mangler {

    private Mangler() throws IllegalAccessException {
        throw new IllegalAccessException("This class cant be instantiated");
    }

    public static String mangle(String key, int quantity) {
        return Mangler.class.getName() + "_key_" + key + "_times_" + quantity;
    }

    public static String mangle(String key) {
        return Mangler.class.getName() + "_key_" + key;
    }

}

File: loquacious/src/main/java/com/saantiaguilera/loquacious/util/Mangler.kt
package com.saantiaguilera.loquacious.util

import com.saantiaguilera.loquacious.model.Quantity

/**
 * Created by saguilera on 11/18/17.
 */
class Mangler @Throws(IllegalAccessException::class) private constructor() {

    init {
        throw IllegalAccessException("This class cant be instantiated")
    }

    companion object {

        private fun clean(string: String) = string.replace(Regex("[^a-zA-Z0-9]"), "_")

        fun mangle(key: String, quantity: Quantity): String = "key_" + clean(key) + "_times_" + quantity

        fun mangle(key: String): String = mangle(key, Quantity.ONE)

    }

}
